src/calc/phased_payment_calculator.py

```diff
@@ -127,6 +132,32 @@ def calculate(self, reward_provider_model, adjustments=None):
             rwrd_logs, total_rwrd_amnt, adjustments
         )
 
+        # run phases again if calculated minimal payment amount is greater than configured
+        if self.min_payment_amnt and self.min_payment_amnt > int(
+            min([rl.adjusted_amount for rl in rwrd_logs if not rl.skipped])
+        ):
+            self.min_delegation_amnt = int(
+                min(
+                    [
+                        rl.staking_balance
+                        for rl in rwrd_logs
+                        if not rl.skipped
+                        and rl.adjusted_amount > self.min_payment_amnt
+                        and rl.staking_balance
+                    ]
+                )
+            )
+            logger.info(
+                "Setting min_delegation_amt to {:<,d} mutez due to min_payment_amt set to {:<,d}. Running calculations again.".format(
+                    self.min_delegation_amnt, self.min_payment_amnt
+                )
+            )
+            rwrd_logs, total_rwrd_amnt = self.calculate(
+                reward_provider_model, adjustments, True
+            )
+        elif rerun:
+            return rwrd_logs, total_rwrd_amnt
```

Comment:
Please remove the return statement and the rerun variable since it is redundant. The function calculate returns anyway with reward_logs and the total reward amount. Ofc it sorts and logs in between but it should be fine.